fabtests/regression/sighandler_test.c

```diff
@@ -66,9 +67,19 @@ int main(int argc, char **argv)
 		if (!hints)
 			exit(EXIT_FAILURE);
 
-		hints->fabric_attr->prov_name = strdup("shm");
+		while ((op = getopt(argc, argv, "p:h")) != -1) {
+			switch (op) {
+			case 'p':
+				hints->fabric_attr->prov_name = strdup(optarg);
+				break;
+			case '?':
+			case 'h':
+				FT_PRINT_OPTS_USAGE("-p <provider>", "specific provider name eg shm, efa");
+				return EXIT_FAILURE;
+			}
+		}
 		if (ft_init_fabric()) {
-			fprintf(stderr, "Failed to initialize shm provider\n");
+			fprintf(stderr, "Failed to initialize %s provider\n", hints->fabric_attr->prov_name);
```

Comment:
line length is long.  If the user doesn't specify the -p option, I think hints->fabric_attr->prov_name will be NULL.

Reply:
No, I don't think the print is necessary; and as written I think can produce a segfault.  There should be some sort of print closer to the actual error, somewhere inside the ft_init_fabric() call, if needed.

Reply:
Thanks, I removed this printing in the new revision then.

Reply:
Just a note: printf("%s", NULL) would produce "(null)" and won't crash.

Reply:
I didn't know they made printf smart enough to do that.  Good to know.